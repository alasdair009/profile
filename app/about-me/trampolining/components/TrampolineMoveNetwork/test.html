<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
