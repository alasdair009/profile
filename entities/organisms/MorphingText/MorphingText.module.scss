@use "../../../app/styles/dimensions";
@use "../../../app/styles/typography";
@use "../../../app/styles/helpers/rem";

@keyframes wordAnim {
  0%,
  5%,
  100% {
    filter: blur(0);
    opacity: 1;
  }
  20%,
  80% {
    filter: blur(1em);
    opacity: 0;
  }
}

.root {
  display: flex;
  margin: 0 auto;
  max-width: rem.toRem(dimensions.$globalContentMaxWidth);
  position: relative;
  width: 100%;
}

.text {
  --speed: calc(var(--word-count) * 1.5 + 2);
  --delay: calc(
    ((var(--speed) / (var(--word-count) + 1))) *
      var(--word-index) - var(--speed)
  );
  animation: wordAnim calc(var(--speed) * 1s) infinite ease-in-out;
  animation-delay: calc(var(--delay) * 1s);
  display: inline-block;
  font-size: typography.getFontSize(xxlarge) !important;
  left: 50%;
  position: absolute;
  text-align: center;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;

  @media (prefers-reduced-motion) {
    animation: none;
    left: auto;
    position: relative;
    top: auto;
    transform: none;
  }
}

.morpher {
  border: none !important;
  filter: contrast(25) blur(rem.toRem(1));
  position: relative;
  min-height: rem.toRem(480);
  width: 100%;

  @media (prefers-reduced-motion) {
    flex-direction: column;
    display: flex;
    justify-content: center;
  }
}
