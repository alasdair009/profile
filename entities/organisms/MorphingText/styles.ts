"use client";

import styled, { keyframes } from "styled-components";
import { fontSizes, globalContentMaxWidth, Heading } from "@/entities";
import { rem } from "polished";

const wordAnim = keyframes`
    0%, 5%, 100% { filter: blur(0); opacity: 1; }
    20%, 80% { filter: blur(1em); opacity: 0; }
`;

const getSpeed = function (wordCount: number) {
  return wordCount * 1.5 + 2;
};
const getDelay = function (wordCount: number, wordIndex: number) {
  const speed = getSpeed(wordCount);
  return (speed / (wordCount + 1)) * wordIndex - speed;
};
export const Root = styled.section`
  display: flex;
  margin: 0 auto;
  max-width: ${rem(globalContentMaxWidth)};
  position: relative;
  width: 100%;
`;

export const Text = styled.span<{ $wordIndex: number; $wordCount: number }>`
  animation: ${wordAnim} ${({ $wordCount }) => getSpeed($wordCount)}s infinite
    ease-in-out;
  animation-delay: ${({ $wordIndex, $wordCount }) =>
    getDelay($wordCount, $wordIndex)}s;
  display: inline-block;
  font-size: ${fontSizes.xxlarge.rem};
  left: 50%;
  position: absolute;
  text-align: center;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;

  @media (prefers-reduced-motion) {
    animation: none;
    left: auto;
    position: relative;
    top: auto;
    transform: none;
  }
`;

export const Morpher = styled.h2`
  filter: contrast(25) blur(${rem(1)});
  position: relative;
  min-height: ${rem(480)};
  width: 100%;

  @media (prefers-reduced-motion) {
    flex-direction: column;
    display: flex;
    justify-content: center;
  }
`;
