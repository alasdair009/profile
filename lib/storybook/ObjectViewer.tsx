import { Source } from "@storybook/blocks";

import type { JSX } from "react";

export type ObjectViewerProps = {
  /**
   * The object to display
   */
  object: object;
  /**
   * The title of the code segment
   */
  title: string;
};

export function ObjectViewer({
  object,
  title,
}: ObjectViewerProps): JSX.Element {
  // Format the object as JSON with indentation
  const objectAsString = JSON.stringify(object, null, 4);

  return (
    <div title={title} data-testid={ObjectViewer.name}>
      <Source code={objectAsString} dark={true} />
    </div>
  );
}
