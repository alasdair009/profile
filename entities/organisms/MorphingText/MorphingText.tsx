import { HTMLAttributes } from "react";
import { Root, Text, Morpher } from "./styles";

type MorphingTextProps = {
  values: string[];
} & HTMLAttributes<HTMLDivElement>;

export function MorphingText({ values, ...rest }: MorphingTextProps) {
  return (
    <Root data-testid={MorphingText.name} {...rest}>
      <Morpher>
        {values.map((value, index) => {
          return (
            <Text
              $wordCount={values.length}
              $wordIndex={index}
              key={value}
              data-testid={`${MorphingText.name}Text`}
            >
              {value}
            </Text>
          );
        })}
      </Morpher>
    </Root>
  );
}
