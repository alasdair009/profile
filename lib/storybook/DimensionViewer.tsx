import { Source } from "@storybook/blocks";
import type { HTMLAttributes } from "react";

import { ObjectViewer } from "./ObjectViewer";
import { colors } from "@/entities";

export type DimensionViewerProps = {
  /**
   * The units to display
   */
  units: Record<string, unknown>;
  /**
   * A simplified object that can be used for the DimensionBar interface
   */
  barUnits: Record<string, string | number>;
  /**
   * The title of the code segment
   */
  title: string;
  /**
   * The suffix to display after the unit
   */
  suffix?: string;
  /**
   * Source example
   */
  sourceExample: string;
  /**
   * The name of the object
   */
  objectName: string;
};

export function DimensionViewer({
  units,
  title,
  sourceExample,
  objectName,
  barUnits,
  suffix = "",
}: DimensionViewerProps): JSX.Element {
  return (
    <div data-testid={DimensionViewer.name}>
      <ObjectViewer object={units} title={objectName} />
      <p>
        To use these values import the <strong>{objectName}</strong> object and
        reference the required key.
      </p>
      <Source code={sourceExample} dark={true} language="css" />
      <DimensionBarWrapper title={title}>
        {Object.keys(barUnits).map((key, index) => {
          return (
            <DimensionBar
              key={`${title}${index}`}
              data-testid={DimensionViewer.name}
              width={`${barUnits[key]}${suffix}`}
            >
              <DimensionBarLabel>
                {barUnits[key]}
                {suffix}
                <br />({key})
              </DimensionBarLabel>
            </DimensionBar>
          );
        })}
      </DimensionBarWrapper>
    </div>
  );
}

const DimensionBarWrapper = ({ children }: HTMLAttributes<HTMLDivElement>) => (
  <div
    style={{
      display: "flex",
      flexDirection: "column",
      overflow: "scroll",
      width: "100%",
    }}
  >
    {children}
  </div>
);

const DimensionBar = ({
  children,
  width,
}: { width: string } & HTMLAttributes<HTMLDivElement>) => (
  <div
    style={{
      background: colors.greenGrass,
      color: "white",
      flex: 1,
      marginBottom: "8px",
      width,
    }}
  >
    {children}
  </div>
);

const DimensionBarLabel = ({ children }: HTMLAttributes<HTMLSpanElement>) => (
  <span
    style={{
      display: "block",
      marginLeft: "8px",
    }}
  >
    {children}
  </span>
);
