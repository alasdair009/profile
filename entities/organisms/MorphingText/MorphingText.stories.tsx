import { Meta, StoryObj } from "@storybook/nextjs";
import { MorphingText } from "./MorphingText";
import { expect, within } from "storybook/test";

const meta: Meta<typeof MorphingText> = {
  component: MorphingText,
  args: {
    values: ["The", "possibilities", "are", "endless", "1", "2", "3"],
  },
  parameters: {
    controls: { hideNoControlsWarning: true },
  },
};
export default meta;

export const Default: StoryObj<typeof MorphingText> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const morphingTextElement = canvas.getByTestId(MorphingText.name);
    const textElements = canvas.getAllByTestId(`${MorphingText.name}Text`);
    const firstTextElement = canvas.getByText(args.values[0]);
    const lastTextElement = canvas.getByText(
      args.values[args.values.length - 1]
    );

    await expect(morphingTextElement).toBeInTheDocument();
    await expect(textElements).toHaveLength(args.values.length);
    await expect(firstTextElement).toBeInTheDocument();
    await expect(lastTextElement).toBeInTheDocument();
  },
};
