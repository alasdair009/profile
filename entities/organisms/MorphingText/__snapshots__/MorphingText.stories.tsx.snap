// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/MorphingText Default smoke-test 1`] = `
<section class="sc-ezTrPE dnqBjY">
  <h2 class="sc-hDcvty YRVwn">
    <span class="sc-drnuxz kEkKRK">
      The
    </span>
    <span class="sc-drnuxz iBrPjB">
      possibilities
    </span>
    <span class="sc-drnuxz kHjzAg">
      are
    </span>
    <span class="sc-drnuxz emCIDl">
      endless
    </span>
    <span class="sc-drnuxz ksVpvZ">
      1
    </span>
    <span class="sc-drnuxz jZfYls">
      2
    </span>
    <span class="sc-drnuxz jIUDvF">
      3
    </span>
  </h2>
</section>
`;
